app/assets/javascripts/govuk_publishing_components/load-analytics.js

```diff
@@ -71,38 +73,119 @@ window.GOVUK.loadAnalytics = {
+      }
+
+      window.GOVUK.analyticsGa4.vars.internalDomains = []
+      window.GOVUK.analyticsGa4.vars.internalDomains.push(window.GOVUK.analyticsGa4.core.trackFunctions.getHostname())
+      window.GOVUK.analyticsGa4.core.trackFunctions.appendDomainsWithoutWWW(window.GOVUK.analyticsGa4.vars.internalDomains)
+    }
+  },
+
+  addGtmScriptToDom: function () {
+    var firstScript = document.getElementsByTagName('script')[0]
+    var newScript = document.createElement('script')
+    newScript.async = true
+
+    // initialise GTM
+    window.dataLayer = window.dataLayer || []
+    window.dataLayer.push({ 'gtm.blocklist': ['customPixels', 'customScripts', 'html', 'nonGoogleScripts'] })
+    window.dataLayer.push({ 'gtm.start': new Date().getTime(), event: 'gtm.js' })
+
+    var auth = window.GOVUK.analyticsGa4.vars.auth || ''
+    var preview = window.GOVUK.analyticsGa4.vars.preview || ''
+    if (auth) {
+      auth = '&gtm_auth=' + auth
+    }
+    if (preview) {
+      preview = '&gtm_preview=' + preview + '&gtm_cookies_win=x'
+    }
+
+    this.googleSrc = 'https://www.googletagmanager.com/gtm.js?id=' + window.GOVUK.analyticsGa4.vars.id + auth + preview
+    newScript.src = this.googleSrc
+    firstScript.parentNode.insertBefore(newScript, firstScript)
+  },
+
   loadGa4: function (currentDomain) {
     currentDomain = currentDomain || window.location.hostname
-    var environment = ''
 
+    window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {}
+
+    if (typeof window.GOVUK.analyticsGa4.core === 'undefined') {
+      console.error('load-analytics.js: window.GOVUK.analyticsGa4.core was not found - has ./analytics-ga4.js been imported?')
+      return
+    }
+
+    this.environment = ''
     // Categorise current environment
     if (this.arrayContains(currentDomain, this.productionDomains)) {
-      environment = 'production'
+      this.environment = 'production'
     } else if (this.arrayContains(currentDomain, this.stagingDomains)) {
-      environment = 'staging'
+      this.environment = 'staging'
     } else if (this.arrayContains(currentDomain, this.integrationDomains)) {
-      environment = 'integration'
+      this.environment = 'integration'
     } else if (this.arrayContains(currentDomain, this.developmentDomains) || currentDomain.indexOf('.dev.gov.uk') !== -1) {
-      environment = 'development'
+      this.environment = 'development'
+    }
+
+    if (!this.environment) {
+      console.error('load-analytics.js: environment for GA4 not recognised - ' + this.environment + ' on ' + currentDomain)
+      return
+    }
+
+    // If the current environment has initialiseGA4 = false, stop loading GA4
+    if (!this.ga4EnvironmentVariables[this.environment].initialiseGA4) {
+      return
+    }
+
+    // to be added: digital identity consent mechanism
+    var consentCookie = window.GOVUK.getConsentCookie()
+    if (consentCookie && consentCookie.usage) {
```

Comment:
Non blocking - I worry that this section is now slightly duplicating the equivalent section for UA i.e. checking for cookie consent before doing anything. Obviously we'll get rid of that other code at some point but I wonder if there's a way of combining them again to save checking for cookie consent twice.

Reply:
@andysellick Good idea - the UA initialisation code looks a bit more complicated though so combining them may take a bit of time if we decided to do this.